### DISCUSSION TITLE
Why do we suggest disallow squash/rebase commits?

### DISCUSSION THREAD

**Discussion by KennethEnevoldsen** [MartinBernstorff/swift-python-cookiecutter#66] [General]
from the `recommended_repo_setup.md`

>     * Disallow squash commits 
>     * Disallow rebase commits

Also maybe we should move `recommended_repo_setup.md` to the readme (so outside the template)?

**Comment by MartinBernstorff** [comment]
Re: Also maybe we should move recommended_repo_setup.md to the readme (so outside the template)?, have responded in the PR where you implemented it ;-)

No deeper reasoning, just preference.

Squash: **Much** prefer to have higher granularity for `git bisect` if debugging in the future, but fine with removing.

Rebase: Very frequent merge errors makes it more complicated, but fine with leaving it open.

**Reply by KennethEnevoldsen** [reply]
Okay so it seems strict to recommend it rather than just leave it as a preference of the user?

**Reply by MartinBernstorff** [reply]
I would disallow squash on all my repos because of the reasoning above. It's just a reminder to myself; don't think the recommendation is too strict ;-)

**Reply by KennethEnevoldsen** [reply]
You are probably right; they are just recommendations/reminders.
